packages/puppeteer-core/src/bidi/Frame.ts

```diff
@@ -106,7 +106,7 @@ export class BidiFrame extends Frame {
     this.browsingContext.on('closed', () => {
       for (const session of BidiCdpSession.sessions.values()) {
         if (session.frame === this) {
-          void session.detach().catch(debugError);
+          session.onClose();
```

Comment:
If the `BrowserContext` is closed Mapper the session is already closed so we need to only remove it from the map.